blueocean-rest-impl/src/test/java/io/jenkins/blueocean/service/embedded/rest/ArtifactImplTest.java

```diff
@@ -20,6 +21,7 @@ public class ArtifactImplTest {
     @Test
     public void findUniqueArtifactsWithSameName() throws IllegalAccessException, NoSuchFieldException {
         //mock artifacts
+        assumeTrue("TODO in Java 12+ final cannot be removed", Runtime.version().feature() < 12);
```

Comment:
This test relies on the use of reflection, which no longer works as of Java 12+.